gems/sorbet-runtime/lib/types/private/methods/decl_builder.rb

```diff
@@ -127,15 +128,27 @@ def abstract
         raise BuilderError.new("`.abstract` cannot be combined with `.override` or `.overridable`.")
       end
 
+      if blk
+        T::Private::DeclState.current.active_declaration.blk = blk
+      end
+
       self
     end
 
-    def final
+    def final(&blk)
       check_live!
-      raise BuilderError.new("The syntax for declaring a method final is `sig(:final) {...}`, not `sig {final. ...}`")
+
+      active_declaration = T::Private::DeclState.current.active_declaration
+
+      if blk
+        active_declaration.blk = blk
+        active_declaration.final = true
+      end
+
+      self
```

Comment:
similar to the static error below, i think we should still have a runtime error if someone attempts to do

```
sig {final.void}
```

and only allow `sig.final {void}` in sorbet-runtime.